modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -1158,71 +1167,68 @@ public void updateStateIndex(Experiment experiment) {
     }
 
     protected void updateStateIndex(Experiment.ID experimentID, ExperimentState state) {
+        driver.getSession().execute(updateStateIndexStatement(experimentID, state));
+    }
 
+    protected BatchStatement updateStateIndexStatement(Experiment.ID experimentID, ExperimentState state) {
         LOGGER.debug("update state index experiment id {} state {} ", new Object[]{experimentID, state});
+        BatchStatement batchStmt = new BatchStatement();
 
         try {
             switch (state) {
                 case DELETED:
                     LOGGER.debug("update state index insert experiment id {} state {} ",
                             new Object[]{experimentID, ExperimentState.DELETED.name()});
 
-                    BatchStatement batch1 = new BatchStatement();
                     Statement insertStatement1 = stateExperimentIndexAccessor.insert(ExperimentState.DELETED.name(),
                             experimentID.getRawID(), ByteBuffer.wrap("".getBytes()));
-                    batch1.add(insertStatement1);
+                    batchStmt.add(insertStatement1);
```

Comment:
@bargenilesh Can we please rename these statement variables more appropriately instead of `insertStatement1` , `deleteStatement1` etc. as part of this change itself as we are anyway reintroducing this entire code as a new function. Code refactoring on the go! :)